% Abstract.tex


The {\it C. acetobutylicum} genome annotation has been markedly improved by integrating bioinformatic predictions with RNA sequencing data. An analysis of an initial assembly revealed errors due to technical and biological background signals, issues which have not been sufficiently addressed in the genomic literature. Additional hurdles for transcriptome research using RNA-seq include both library complexity and depth, yet most studies on this topic in bacteria report low depths and do not account for ribosomal RNA abundance. In this work, in vitro and in silico workflows were established to ensure data quality. A cumulative median depth of 156 was observed throughout the genome after rRNA filtering, facilitating the initial transcriptome assembly. The assembly was then contextualized to provide precise and accurate determination of gene boundaries. To this end, a visualization tool was designed to integrate motif predictions, nucleotide resolution coverage, and the transcriptome assembly for the purpose of assembly curation. The resulting expression and annotation can be readily explored and downloaded, enabling future genomic and transcriptomic research in this organism.

