% Abstract.tex


The {\it C. acetobutylicum} genome annotation has been markedly improved by integrating bioinformatic predictions with RNA sequencing(RNA-seq) data. Analysis of an initial assembly revealed errors due to technical and biological background signals, challenges with few solutions in the genomic literature. Additional hurdles for next-generation sequencing(NGS) based transcriptome mapping research include optimizing library complexity and sequencing depth, yet most studies in bacteria report low depth and ignore the effect of ribosomal RNA abundance and other sources on the effective sequencing depth. 

In this work, \textit{in vitro} and \textit{in silico} workflows were established to address false positive and negative errors associated with transcriptome mapping. An integrative analysis method was developed to integrate motif predictions, single-nucleotide resolution sequencing depth, and complexity during curation. This contextualization minimized false positive error, providing the precise and accurate determination of gene boundaries qualitifed by previous studies, in some cases, to the exact basepair. Curation of the pSOL1 megaplasmid improved statistical measures of transcriptomic features to be amenable with findings from \textit{E. coli}. 

The resulting annotation can be readily explored and downloaded through a customized genome browser, enabling future genomic and transcriptomic research in this organism. This work demonstrates the first strand-specific transcriptome assembly in the \textit{Clostridia}. Additionally, this method can be used to eliminate false positive features from assemblies in bacterial transcriptome mapping studies. 