
%%% Local Variables: 
%%% mode: latex
%%% TeX-master: "main"
%%% End: 



\section{Example Transcripts and Curation Process}
Previous sections described global indicators of misassemblies cause by background signals and the tools required to identify and address them. Correction also required clear description of these errors and a defined curation method, facilitated by the genome browser and genomic signals. These errors, not discussed or documented in similar studies, were best described for genes that have previously determined transcript boundaries. This external validation allowed true understanding of the type I and type II errors in the uncurated assembly. Six issues, listed below, were considered for each example to better understand the quality of the assembly and the degree of curation required. 

\begin{enumerate}
\item Was the transcript large enough to include the known ORFs and RBSes?
\item Did the assembled transcript's TSS agree with promoter motifs?
\item Did it agree with published transcription start sites?
\item Did the assembled transcript's size agree with published Northern blots?
\item Did the assembly represent the coverage and if not, which of these two best represents the biological knowledge of this region?
\item Did the assembled region require curation (e.g. fused, extended, or truncated transcripts)?
\end{enumerate}

These questions were considered for each of 5 loci where there is \textit{a priori} knowledge. In the following examples, the uncurated assembly results were compared to the promoter, terminator, and sequencing data for the region. Misassemblies were documented when the assembled transcript disagreed with obvious patterns in sequencing depth, promoter, and terminator annotations. These examples illustrated the types of errors found in the data and details the simple methods to correct them. The first example, the Sol locus, contains 3 transcripts that display minor extension misassemblies with clear and simple solutions. 

\subsection{Sol Locus}

The \textit{sol} locus is a 7kb region on the pSOL1 megaplasmid surrounding the \textit{sol} operon(4.3kb, basepairs 175,564-179,841). This region is responsible for the production of several solvents\cite{62,63} and is immensely important to the physiology of the \textit{C. acetobutylicum} ATCC 824 strain. This region encodes several enzymes including a tri-functional NAD(H\textsuperscript{+})-dependent alcohol/aldehyde dehydrogenase (AdhE1)\cite{62}, two subunits of coenzyme-A transferases (CtfA/B)\cite{66}, and an acetoacetate decarboxylase (Adc)\cite{64,65,66}. The region is also home to a protein SolR, which includes a helix-turn-helix motif and is thought to regulate solventogenesis\cite{67}. These genes are vital for carboxylic acid reuptake and conversion into alcohols, a vital part of this organism's metabolism and the solventogenesis process.

%          SOL LOCUS Fig 1.
\begin{figure}
\small
{\includegraphics[width=\textwidth,height=2.5in]{images/Assembly/Examples/Sol/Sol-locus-curated.png}
\subcaption{Sol locus}\label{fig:5.9}}
% \label{fig:1}
\caption{Sol Locus Overview} This operon (upper track) consists of orfL, alcohol dehydrogenase (adhE1), and Co-A transferases A and B (ctfA,ctfB). solR (far left) and acetoacetate decarboxylase (adc; lower track, right) are also shown. Coverage for the Watson and Crick strands (top and bottom tracks) are visualized with an annotation track (center). Tracks show cumulative coverage for unstressed (yellow), butanol (light green/ light orange), and butyrate (green/orange) stressed samples over all time points. Transcripts (blue), ORFs (orange), RBSes (purple), inverted repeats (yellow), promoters (green), and TSSes (red) are represented as arrows and bars.
\end{figure}

\subsubsection{Acetoacetate Decarboxylase Transcript}
In the early 1990s, several articles were published about the \textit{sol} locus including the cloning and sequencing of adc and the \textit{sol} locus\cite{62,63,64,65,66}. An early study of the \textit{sol} operon probed the adc locus, reporting two transcript sizes of 670 and 865 with Northern blot\cite{65}. The authors also reported the major transcription start site of adc at base 180,671 of the pSOL1 plasmid. To examine the quality of our data and raw assembly, we examined this locus to observe the transcript size and locate the transcription start site in our data. 

In \ref{fig:5.10a} we see the transcription start site reported by Durre \textit{et al.}(red)\cite{65} located very near a sustained increase in sequencing coverage just downstream of a canonical promoter motif. This pattern of coverage (cumulatively \textgreater 10,000x) is sustained until a bidirectional Rho-independent terminator (\ref{fig:5.10b}). In this instance, the precise transcription start site was not estimated precisely by the uncurated assembly. The reported transcript continues for several hundred basepairs upstream of the adc TSS, despite the decrease in coverage. This artifact is most likely due to sufficient k-mer complexity in the reads mapping upstream of the TSS for the assembly algorithm to fuse these reads to the adc transcript. While this complexity is generally a good sign for the quality of this dataset, in this case a misassembly was the result. Correcting for this error (\ref{fig:5.10c}), the full transcript size is 857bp.

It was claimed that a 670bp product was most likely a specific degradation product or the result of a secondary transcriptional start site\cite{65}. To investigate this, a transcript of this size would correspond to a transcriptional start site at approximately base 180,484. Unfortunately, none of the promoter motifs in the region could explain a transcript of this size in vegetative cells. After curation based on the coverage pattern, promoter and terminator motifs in this region, the transcription start site and transcript size for adc accurately match previous results. The uncurated assembly predicted a transcription stop site with good precision, in contrast to the start site. The degree of k-mer complexity, despite the low coverage in the background signal upstream of adc, suggests that library complexity is reasonably high. In this instance, the background signal poses and obstacle for automated assembly, but suggests that sufficient depth is achieved in this experiment. This type of misassembly will be referred from here on as an ``extension''. From \ref{fig:5.10b}, another extension misassembly is present in our next example transcript, the \textit{sol} operon.

\begin{figure}
\small
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/Adc-TSS.png}
\subcaption{adc transcription initiation region on the Crick strand.}\label{fig:5.10a}}
{\includegraphics[width=\textwidth,height=2.5in]{images/Assembly/Examples/Sol/Sol-bifunctional-terminator.png}
\subcaption{Bifunctional Rho-independent terminator for \textit{sol} operon (upper track, left) and Adc transcripts}\label{fig:5.10b}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/Adc-curated.png}
\subcaption{Curated adc locus}\label{fig:5.10c}}
\caption{Adc locus}
\subref{fig:5.10a}) Transcription initiation region for adc. While the coverage clearly shows the appropriate increase, the transcription start site has been fused to residual coverage upstream of the true TSS. \subref{fig:5.10b}) A bifunctional terminator is responsible for transcriptional termination of both Adc and the \textit{sol} operon. \subref{fig:5.10c}) With minor curation, the region matches previous results faithfully.
\end{figure}


\subsubsection{Sol Operon}
In \cite{65}, the \textit{sol} operon was investigated using a probe specific for ctfB and a transcript of size of 4.1kb was reported. Unfortunately, no blots were included as figures in this work. In \cite{63} adhE1-based probes revealed a nearly identical transcript size of 4.1-4.2kb. Interestingly, the \textit{sol} operon has both proximal and distal transcription start sites at 175,726 and 175,564, respectively\cite{62,63}. Ribosome binding sites have been identified upstream of adhE1, ctfA, and ctfB\cite{63}. From these previous studies, good recall of the transcription start sites was expected.

The expression level of the \textit{sol} operon is substantial, also upwards of 10,000x coverage cumulatively. The distal transcription start site is matched perfectly (\ref{fig:5.11a}), demonstrating the precision of the assembly technique in the absence of background or residual signal. An increase in coverage is observed immediately following the proximal promoter (\ref{fig:5.11a}) in close agreement with the previous determinations\cite{62,63}. However, the transcription stop site was not precisely determined by the assembly, owing in part to basal antisense signal from the Adc gene (\ref{fig:5.10b}). After adjustment, the transcript sizes are 4,115 and 4,277, respectively, in close agreement with the reported transcript sizes\cite{63,65}.

\begin{figure}
\small
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/Sol-TSS.png}
\subcaption{\textit{sol} operon transcription initiation region. The distal (left) and proximal (right) transcription start sites (red) are shown for adhE (far right, orange).}\label{fig:5.11a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/AdhE-terminator.png}
\subcaption{Putative adhE1 (left) terminator, ctfA (right) promoter}\label{fig:5.11b}}
\caption{Sol Operon}
\subref{fig:5.11a}) \textit{sol} operon (orfL, center; adhE right) transcription start sites. The coverage and assembly data have strong agreement with previously described proximal and distal promoters and transcription start sites. \subref{fig:5.11b}) Low coverage in the \textit{sol} operon. A terminator may be partially responsible for a sustained low coverage level in the \textit{sol} operon. Additionally, a promoter motif was located upstream of the ctfA RBS and the pattern of expression is consistent with these observations.
\end{figure}

\paragraph{Multiple Transcripts from Sol Operon}
While the results from this region agree as a whole, there is an interesting pattern in coverage in the \textit{sol} operon near the C-terminus of adhE1(\ref{fig:5.11b}). This 100bp region is expressed at a statistically lower level (K.S.-test, p < 0.05) than the rest of the \textit{sol} operon but has a standard GC content of 35\%. It is unlikely that the low sequencing depth is caused by sequence specific biases, described earlier. Upon further examination of this region, we find a Rho-independent terminator with a \(\Delta\)G of -9.6 kcal/mol that is not as strong as the -11.5 kcal/mol bifunctional terminator at the end of the \textit{sol} operon. The region is also near a $\sigma_{A}$ promoter motif of TTCATA(13)TATAAT located upstream of the previously mentioned RBS. 

As mentioned above, no Northern blots figures were included in the only study, to the best of my knowledge, that uses ctfA or ctfB specific probes\cite{65}. Most studies of the \textit{sol} operon in \textit{C. acetobutylicum} use adhE1-specific probes or larger restriction digestion probes \cite{63,68,71}. One of these displays a Northern with a weak but distinct band for a ~2.6kb transcript under solventogenic conditions \cite{69}. If adhE1 were transcribed both in the classical \textit{sol} operon and as a separate transcript, the length would be 2,716bp from the proximal transcription start site, similar in size to the observed band\cite{69}. 

This region shows steep changes in coverage that are consistent across all replicates. It is likely that this observation does not merely represent difficulty sequencing secondary structure in the \textit{sol} operon transcript. In addition to the full \textit{sol} operon transcript, additional transcripts from the AdhE and ctfA/B genes may be produced. Recent Northern blots from our lab seem to corroborate this finding (Alex Jones, personal communication, December 9, 2014).

In addition to correcting misassembly, identifying overlapping transcripts and alternative transcription start sites is a benefit of curation. The uncurated Sol transcript has basepair-level resolution of the distal transcription start site. With minor curation, two additional transcripts were discovered from this region and the stop site was easliy determined. For the first two example transcripts, boundaries were determined with high precision and accuracy, although it should be noted that these regions had strong expression. In the next example, high precision is achieved again, even in a transcript with lower coverage.


\subsubsection{SolR Transcript}
The last transcript produced from the \textit{sol} locus considered here is from the solR gene. This gene produces two transcripts, one 1kb and the other 1.3kb\cite{63}. A later study revealed the role of SolR as a repressor for the \textit{sol} operon\cite{69}. This study also produced a single transcription start site at 174,154 on the pSOL1 plasmid. As a result of examining this dataset, perfect recapitulation of the transcription start site was achieved(\ref{fig:5.12a}). The transcript from the raw assembly is approximately 1.2kb(\ref{fig:5.12b}), showing only residual coverage (cumulatively \textless 5x) after the first terminator(\ref{fig:5.12c}). After curation (\ref{fig:5.12d}), the transcript size is 1,036bp. There was insufficient coverage in the conditions studied to strongly support the larger transcript. No alternative transcription start sites were apparent.

In the first locus, boundaries were accurately determined after minor curation. Examples of the first type of misassembly, the ``extension'' were presented. The level of complexity of the reads mapping to these regions was high, given complete assembly of these regions including low coverage background signal. Basepair level resolution of the transcription start sites was observed for 2 of the 3 transcription start sites without curation. With the misassemblies that were described here, minor curation was required to accurately determine the remaining boundaries. In the next example locus, two butanol dehydrogenase transcripts will be examined, along with the next type of misassembly that this curation approach can address.


% Transcription start sites SolR
\begin{figure}

{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/SolR-TSS.png}
\subcaption{solR Transcription Initiation Region}\label{fig:5.12a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/SolR-transcript.png}
\subcaption{solR Transcript}\label{fig:5.12b}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/SolR-termination.png}
\subcaption{solR Transcription Termination Region}\label{fig:5.12c}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Sol/SolR-curated.png}
\subcaption{Curated solR Transcript}\label{fig:5.12d}}
%\label{fig:1.5}
\caption{SolR Locus}
\subref{fig:5.12a}) The assembled transcription start site for solR agrees with previous findings. \subref{fig:5.12b}) The assembled solR transcript has an extended 3' UTR and subref{fig:5.12c}) insufficient coverage for a transcript past the first terminator. \subref{fig:5.12d}) The curated solR transcript agrees with previously published findings\cite{69}.
\end{figure}

\subsection{Bdh Locus}
The bdh locus encodes two homologous butanol dehydrogenase(BDH) enzymes in a 3kb region on the main chromosome. Early studies of butanol dehydrogenases in \textit{Clostridia} located a number of NADH-dependent and NADPH-dependent butanol and alcohol dehydrogenases responsible for butanoate metabolism\cite{70,71,72,73}, specifically the reduction of butyryl and acetyl groups into the solvents butanol and ethanol. One such locus in \textit{C. acetobutylicum} produces two isozymes with different physiological roles. These isozymes like have distinct regulation and physiological roles from the other alcohol dehydrogenases found in this organism. The bdh locus proteins were described by several authors, reporting different activities and specificities for each enzyme\cite{70,71}. After characterizing the enzymes in this locus, the region was cloned and two homologous isozymes were found. The two transcripts originating from these isozymes will demonstrate the precision and accuracy of this technique when compared to primer-extension analyses but also the need for assembly curation that reflects the motifs and coverage pattern of the region. We begin by discussing the first of these, bdhA.
\subsubsection{BdhA}
BdhA is an NADH-dependent butanol dehydrogenase that acts on both butyryl and acetyl groups. Studies suggest that this enzyme has fairly comparable activities with both substrates, with slightly higher activities for butyryl groups\cite{71}. This enzyme was observed to have higher activities at low pH, indicative of its physiological role in the conversion of butyric acid to butanol. The entire locus was sequenced, producing ORFs that exactly matched the bdhA and bdhB isozymes\cite{73}. Northern analysis determined that these genes are transcribed separately and not as an operon. BdhA was found to have a 1.3kb transcript and the transcription start site was mapped through primer-extension to base 3,465,240 on the main chromosome of \textit{C. acetobutylicum}\cite{73}. 

Here we find the transcription start site of bdhA one base upstream at 3,465,241(\ref{fig:5.13b}). The results of transcription start site identification agree well with the upstream Sigma-factor A promoter and the aforementioned start site. The uncurated assembly produces a transcription stop site at base 3,464,329, before the stop codon of BdhA(\ref{fig:5.13c}). The pattern of coverage clearly reflects the Rho-independent terminator nearby. The misassembly could be due to low complexity in the reads mapping to this portion of bdhB, possibly resulting from homology with bdhB. This type of misassembly will be referred to from here on as a ``truncation'' misassembly. The final transcript (\ref{fig:5.13d}) reflects the assembly, coverage pattern, and motifs in this locus, agreeing with the transcription start site and a transcript length of 1,282bp. In this example, the fairly obvious coverage pattern was not reproduced by the assembly, demonstrating the need for some simple curation by integrating knowledge of previous experimental data and genome-wide predictions of promoter and terminator motifs with the coverage pattern and assembly. Next, the paralogous gene BdhB produces a similarly sized transcript.
%        B d h A
\begin{figure}
\footnotesize
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Bdh/Bdh-locus.png}
\subcaption{bdh Locus}\label{fig:5.13a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Bdh/BdhA-TSS.png}
\subcaption{bdhA Transcription Initiation Region}\label{fig:5.13b}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Bdh/BdhA-termination.png}
\subcaption{bdhA Transcription Termination Region}\label{fig:5.13c}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Bdh/Bdh-curated.png}
\subcaption{Curated bdhA Transcript}\label{fig:5.13d}}
\caption{Bdh Locus}
\subref{fig:5.13a}) The bdh locus displays an obvious coverage pattern for two monocistronic transcripts. \subref{fig:5.13b}) The bdhA transcript displays a sharp increase in coverage near the transcriptional start site. This data agrees to a good extent with primer extension studies for this gene. \subref{fig:5.13c}) The raw assembly has failed to recapitulate the transcription termination region, likely due to low complexity coverage of the 3' region of this transcript. \subref{fig:5.13d}) The curated transcript reflects experimental characterization of this transcript\cite{73}.
\end{figure}

\subsubsection{BdhB}
The next gene is bdhB, another NADH-dependent butanol dehydrogenase, with a slightly longer transcript size of 1.35kb. It was reported that this enzyme has 46-fold higher activity with butyryl groups than acetyl groups\cite{71,73}. BdhB was sequenced an analyzed along with bdhA, where it was discovered that bdhB had at least two transcription start sites, independent from BdhA. The most dominant transcription start site was very close to a secondary band at approximately 3,463,816 and 3,463,811, respectively\cite{73}. I will refer to these two distal bands collectively as the primary transcription start site for bdhB. A third band was located slightly farther upstream at 3,463,803\cite{73}. 

The coverage pattern for this region shows at least 3 increases in coverage at 3,463,802, 3,463,813, and 3,463,843 (\ref{fig:5.14a}). The first (proximal) site has a promoter motif upstream, corresponding to the 1$^{st}$ -10 and -35 boxes in \ref{table:1} and the tertiary transcription start site from above\cite{72}. The second site has a significant increase in coverage between residue 3,463,817 and 3,463,810, which match well with the primary transcription start site described above\cite{73}. The primary transcription start site matches the 2$^{nd}$  -35 box and the 2$^{nd}$ or 3$^{rd}$ -10 box motifs\cite{73}. It is clear that the strongest start site is the primary transcription start site previously described\cite{73}, although the multiple bands observed in their analysis could indeed be explained by the additional matches to consensus Sigma-factor A motifs. Additionally, an observable but insignificant increase is correlated with a final promoter motif. It is clear that the raw data match previous results to a good extent but curation was required to correct the transcription start site(\ref{fig:5.14a}) and stop site (\ref{fig:5.14b}). The final transcript is shown in \ref{fig:5.13c} with a final length of 1,367bp or 1,381bp, in close agreement with the published length of 1.35kb.

\begin{table}

%\begin{minipage}[b]{2.5in}
\begin{center}
\begin{tabular}{|c|c|c|c|c|}\hline
\multicolumn{5}{c}{-35 box}\\\hline
Motif & Start & End & Sequence & p-value\\\hline
1 & 3463831 & 3463836 & TAGGTT & 3.5\e{-2}\\
2 & 3463847 & 3463852 & TTGTAA & 9.4\e{-3}\\
3 & 3463870 & 3463875 & TGGATA & 2.6\e{-2}\\\hline
\end{tabular}
\end{center}
%\end{minipage}
%\begin{minipage}[b]{\texit}
\begin{center}
\begin{tabular}{|c|c|c|c|c|}\hline
\multicolumn{5}{c}{-10 Box}\\\hline
Motif & Start & End & Sequence & p-value\\\hline
1 & 3463816 & 3463821 & TATAAT & 4.3\e{-4}\\
2 & 3463820 & 3463825 & TATATA & 1.6\e{-3}\\
3 & 3463830 & 3463835 & TAAAAT & 4.2\e{-3}\\
4 & 3463852 & 3463857 & TATTAT & 4.2\e{-3}\\\hline

\end{tabular}
\caption{BdhB Sigma-factor A boxes}\label{table:1}
\end{center}
%\end{minipage}
\end{table}
\begin{figure}
\small
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Bdh/BdhB-TSS.png}
\subcaption{bdhB Transcription Initiation Region}\label{fig:5.14a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Bdh/BdhB-termination.png}
\subcaption{bdhB Transcription Termination Region}\label{fig:5.14b}}
\caption{Bdh Locus and Transcription Start Sites}
\subref{fig:5.14a}) The bdhB transcript has several promoter motifs and matching increases of coverage. These increases agree with previous experimental results\cite{73}, 2 which were dismissed after not identifying the appropriate promoters. Unfortunately, the transcription start site was incorrect in the raw assembly. \subref{fig:5.14b}) A Rho-independent terminator is found at the end of the bdhB transcript although residual coverage triggered misassembly.
\end{figure}

In this example, the bdhA and bdhB transcripts were very close to capturing the true boundaries of these genes. In the case of bdhA the TSS was both precise and accurate, but the assembled transcript did not match the coverage pattern, ORF, and terminator annotations, the first example of a truncation misassembly. It is possible that this and similar misassemblies could be due to homology between paralogues, such as in the C-terminus of bdhA and bdhB. In the case of bdhB, the start and stop sites did not agree with patterns in coverage (extension misassembly) and were simple to correct. The next example is another positive example, this time of a stress-response operon containing the heat-shock proteins GroES and GroEL.

\subsection{GroES/EL Locus}
\subsubsection{GroES/EL Operon}
The GroES and GroEL proteins are evolutionarily conserved heat-shock responsive chaperonins. These proteins are found throughout the tree of life, including \textit{C. acetobutylicum}, where they are an integral to the solvent stress response\cite{74,75,76} and the class I heat-shock response\cite{42,74,75}. Expression levels are both strong and constitutive for this region, with a dramatic and transient increase with solvent or heat stress\cite{74,75}. In a molecular study, GroES and GroEL were found to be produced from a bicistronic operon with a transcript size of 2,150bp\cite{76}. In addition, a Sigma-factor A promoter was identified for an experimentally determined transcription start site. Two bands were observed during the primer-extension assay\cite{76}. The proximal band was dismissed as an artifact although the bands persisted under all conditions examined\cite{76}. This region is known to contain a CIRCE motif upstream\cite{77}, overlapping both of these start sites (\ref{fig:7a})\cite{75}. 

The transcription start site determined in this work is located at 2,829,142, a mere two bases away from a previously reported start site\cite{76} (\ref{fig:5.15a}). Interestingly, the second transcription start site is located at a sharp increase in expression. The distance of this start site from the promoter would suggest either sequencing difficulties or post-transcriptional processing. The transcription stop site is located near a Rho-independent terminator (\ref{fig:5.15b}). The transcript size determined by the raw assembly is 2,131bp in agreement with the 2.2kb band and the 2,150bp calculated distance between the transcription start-site and the Rho-independent terminator\cite{77}. In this example, the uncurated assembly produced a flawless recapitulation of the coordinates and size of the groES/EL operon (\ref{fig:5.15b}). Having established the coordinates for this transcript in agreement with previous findings, the regulation of this operon should be briefly discussed.

\subsubsection{GroES/EL Regulation}
The groES/EL operon is stress responsive, although its expression under stress appears lower in \ref{fig:5.15c}. For this reason, it is worth discussing the regulation of this region and why these results are consistent with knowledge of this area. The CIRCE motif upstream of groES is regulated by HrcA, a heat responsive repressor\cite{42,77,78}. In response to heat-shock, the groES/EL operon is derepressed, revealing a Sigma factor A promoter and resulting in transcription (\ref{fig:5.15a}). The response to heat shock is fairly acute, increasing for 2-3 minutes and returning to standard levels after an additional 10 minutes\cite{77}. Here a general decreased expression is observed in response to solvent stress for two reasons. First, the 6S small RNA is a stress and growth-stage responsive regulator that globally reduces transcription from Sigma factor A promoters\cite{39,79}. Additionally, the time scale assessed here does not include a 3 minute time-point to identify an acute stress response. The stress response observed is the global downregulation of Sigma factor A promoters and the transition of the transcriptome towards one designed to tolerate stress and Sigma factor A dependent promoters are downregulated throughout this dataset as a result of the 6S small RNA. For this reason, we defer the test of differential expression for these time points for the next chapter. In the case of the GroES/EL operon, we observe precise and accurate estimates of transcript boundaries from the uncurated assembly. The next example is the regulator responsible for the acute stress response of these heat shock proteins, HrcA.


\begin{figure}
\small
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/GroESL/GroESL-TSS.png}
\subcaption{GroES/EL Transcription Initiation Region}\label{fig:5.15a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/GroESL/GroESL-termination.png}
\subcaption{GroES/EL Transcription Termination Region}\label{fig:5.15b}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/GroESL/GroESL-curated.png}
\subcaption{GroES/EL Locus}\label{fig:5.15c}}
\caption{GroES/EL Locus}
\subref{fig:5.15a}) groES and groEL form an operon that is responsive to heat-shock through a HrcA-mediated derepression mechanism. The transcription start sites are in agreement with previous findings with the addition of an interesting peak inbetween these two. \subref{fig:5.15b}) The transcription termination region supports previous reports of a 2.2kb transcript terminating at a Rho-independent terminator following the GroEL ORF. 
\end{figure}

%   H r c A,  G r p E,  D n a K,  &  D n a J

\subsection{HrcA and DnaK/J Locus}

\subsubsection{DnaK Locus Overview}
This rather complex locus encodes the class I repressor HrcA, DnaK and DnaJ, another set of evolutionarily conserved heat-shock proteins. DnaK was discovered to be solvent-stress responsive in \textit{C. acetobutylicum}\cite{74,75}. Solvent stress and heat shock increase protein denaturation, requiring molecular chaperones such as GroES/EL and DnaK/J to increase proper protein folding in these conditions. The \textit{C. acetobutylicum} DnaK protein was purified as a stress responsive 74kDa protein\cite{75}. Using a restriction fragment, the DnaK locus was then cloned and sequenced\cite{80}, revealing a grouping of four ORFs, similar in arrangement to \textit{B. subtilis}\cite{77}. An inverted repeat, now known to be the HrcA-binding CIRCE motif, was found upstream of the hrcA gene\cite{80}. Upon Northern analysis of this region, three different transcripts were identified as originating from this locus, of 2.6, 3.8, and 5kb lengths. Additionally, a 1.5kb band was observed with a dnaK specific probe\cite{80}. The first transcript (2.6kb) could be seen with a dnaK-specific probe and is thought to contain the genes grpE and dnaK. The second transcript (3.6kb) was observed for both dnaK and hrcA-specific probes and is thought to contain the same genes plus hrcA. Similarly, the 5kb transcript was observed for both probes and is thought to contain the whole operon. The final transcript(1.4kb) and additional small bands were dismissed as specific degradation products. It has been noted that this operon has interesting post-transcriptional regulation in \textit{B. subtilis}, producing multiple transcripts from a heptacistronic operon\cite{81}. To analyze this complex operon, we will proceed through 4 proposed regulatory sites. The first is the promoter region of hrcA. The second site is a transcription start site upstream of grpE. The third is the location of an internal CIRCE motif, terminator, and an additional transcription start site ahead of dnaJ. The final site is located at the end of the whole operon.

\subsubsection{HrcA Promoter}
The hrcA promoter was described during the sequencing of the dnaK/J locus\cite{80}. This promoter produces the full transcript of 5kb and the smaller 3.6kb transcript terminating between dnaK and dnaJ. Two transcription start sites have been described for this region\cite{80}. Both of the two reported transcription start sites(S1 and S2) were located upstream of the CIRCE element, in contrast to groES/EL (\ref{fig:5.16a}). Additional bands present in the primer extension analysis were rejected similarly to the strand bands in the groES/EL operon. An excellent Sigma factor A motif was located for the S1 site (P1,\ref{table:2}) but only an insignificant motif (p > 0.05, TTTATG(17)AAAGAT) motif was found for the weak band of the S2 site. An alternative promoter (P2, \ref{table:2}) seems to be too close to the S2 transcription start site. If the observed bands represent true transcription start sites for this operon they are transcribed from close and overlapping promoter motifs.

Here we don't see any direct increase in transcription for the S1 site, most likely due sequencing difficulty near the CIRCE motif. Upon TEX enrichment, no increase or decrease in coverage can be observed near these sites(data not shown), suggesting that post-transcriptional processing is not responsible for these transcription start sites. The increases in transcription observed here are relatively minor compared to the transcription of the entire HrcA operon. The P1 and P2 motifs seem to be sufficient for transcription initiation, although the coverage pattern shows evidence of complication of sequencing. In several studies\cite{76,80} transcription start sites have been discarded due to local RNA secondary structure. It is reasonable that reverse transcription in this area may be complicated by the -10kcal/mol hairpin and CIRCE motif near the 5' end of the transcript. 

Several authors have noted that full hrcA/dnaK operon transcripts are present at lower abundance\cite{80,81}, resulting in lower coverage and an indistinct transcription start site when considering coverage alone. In \ref{fig:5.16a}, it is clear that the uncurated assembly estimated the transcription start site more accurately than a coverage-only approach would provide. As we have seen, in some cases the de Bruijn graph assembly requires curation to most accurately reflect local motifs. On the other hand, this method produces a reasonable estimate of the start site. After minor curation, the transcription start site agrees with the previously described\cite{79} S2 and S1 (\ref{fig:5.16b}). Having established the transcription start site for the entire hrcA operon and previous transcriptional start sites are not the result of post-transcriptional processing, we move on to the higher abundance transcripts produced by this region, beginning with the grpE operon.

\subsubsection{GrpE Promoter}
The GrpE protein is an essential nucleotide exchange factor for DnaK. This protein was discovered upstream of dnaK after cloning and sequencing of the dnaK locus\cite{80}. It was postulated that a second transcription start site upstream of grpE would explain the smaller 2.6kb transcript and a transcription start site was determined\cite{80}. A transcription start site exists ahead of the grpE gene in \textit{B. subtilis} as well\cite{81}. The proposed promoter ahead of grpE does not match the Sigma factor A consensus (p > 0.05) and no alternative motifs were found. However, a substantial increase in coverage can be found further upstream from this site. 

The first two sites are specifically enriched in the TEX treated library, suggesting a transcription start site. Promoter motifs of higher quality are found upstream of two major peaks (\ref{fig:5.16b}). Looking at the entire operon (\ref{fig:5.16c}) it is clear that this is a substantial transcription start site. The increase in coverage here is acute in contrast with the previous site in the conditions of this study. Additionally, it has been proposed that the hrcA operon is generated from post-transcriptional processing and differing transcript abundances are due to differing decay rates\cite{81}. If such a processing mechanism were present in \textit{C. acetobutylicum}, the coverage at the grpE transcription start site would be differential with respect to the 5'-phosphate exonuclease (TEX) treatment. In \ref{fig:5.16b}, the coverage is not differential at this or other locations in this transcript, demonstrating that the 2.6kb transcripts is most likely a primary transcript originating from the Sigma factor A promoters P3 or P4.

 In summary, a novel transcription start site and promoter motif were located upstream of grpE. No matching promoter motif or coverage pattern was observed for a previously documented transcription start site under these conditions. Additionally, coverage at the novel transcription start site was enriched with TEX treatment, suggesting that the 2.6kb transcripts are primary transcripts and not products of post-transcriptional processing. CtsR and CIRCE motifs were not found near this transcription start site. Next, the dnaK/J intergenic region is explored, which contains a terminator responsible for read-through transcription of the entire 5kb operon.

\begin{figure}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/HrcA/HrcA-TSS.png}
\subcaption{hrcA Transcription Initiation Region}\label{fig:5.16a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/HrcA/GrpE-TSS.png}
\subcaption{grpE Transcription Initiation Region}\label{fig:5.16b}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/HrcA/HrcA-operon-curated.png}
\subcaption{Curated hrcA Locus}\label{fig:5.16c}}
\caption{HrcA Locus and Promoter Regions}
\subref{fig:5.16a}) HrcA leads the 5kb tetracistronic operon. The regulation of this operon consists of Sigma factor A dependent promoters, a CstR motif, and a CIRCE motif. \subref{fig:5.16b}) A secondary transcription start site in upstream of grpE is responsible for the 2.6kb and 3.8kb transcripts. A large increase in coverage at a novel transcription start site is not TEX responsive. \subref{fig:5.16c}) The curated hrcA operon has two transcription start sites and one Rho-independent terminator which explain the 5kb, 3.8kb, and 2.6kb transcipts reported for this area.
\end{figure}


\begin{table}

%\begin{minipage}[b]{2.5in}
\begin{center}
\begin{tabular}{|c|c|c|c|c|}\hline
\multicolumn{5}{c}{-35 box}\\\hline
Motif & Start & End & Sequence & p-value\\\hline
P1 & 1423790 & 1423795 & TTGACA & 2.9\e{-4}\\
P2 & 1423774 & 1423779 & ATGAAA & 5.3\e{-2}\\
P3 & 1424463 & 1424468 & TTGAGG & 1.6\e{-2}\\
P4 & 1424514 & 1424519 & TTGATT & 6.2\e{-3}\\
P5 & 1427399 & 1427804 & TTGAAA & 2.1\e{-3}\\
\hline
\end{tabular}
\end{center}
%\end{minipage}
%\begin{minipage}[b]{2.5in}
\begin{center}
\begin{tabular}{|c|c|c|c|c|}\hline
\multicolumn{5}{c}{-10 Box}\\\hline
Motif & Start & End & Sequence & p-value\\\hline
P1 & 1423812 & 1423817 & TATTTT & 2.3\e{-2}\\
P2 & 1423800 & 1423805 & TAATGT & 1.8\e{-2}\\
P3.1 & 1424476 & 1424481 & TAATAT & 9.9\e{-3}\\
P3.2 & 1424482 & 1424487 & TAAAAA & 3.2\e{-2}\\
P4 & 1424537 & 1424542 & TATGAT & 1.9\e{-3}\\
P5 & 1427430 & 1427435 & TATAGT & 2.5\e{-3}\\\hline
\end{tabular}
\caption{HrcA Operon Sigma-factor A boxes}\label{table:2}
\end{center}
%\end{minipage}
\end{table}

\subsubsection{DnaK/J Intergenic Region}
The dnaK/dnaJ intergenic region is known to contain a Rho-independent terminator\cite{80,81}, thought to be responsible for the 3.8kb and 2.6kb transcripts. The \(Delta\)G of this terminator is estimated to be -13.2kcal/mol. In \ref{fig:5.17a}, decreased coverage is observed near this terminator, upstream of the dnaJ gene. This terminator does not contain a CIRCE element, in contrast to the inverted repeat at the hrcA promoter. A strong promoter motif (P5,\ref{table:2}) is observed very close to a sharp increase in coverage at the 5' end of the repeat. The previously reported band is found on the 3' end of the repeat, which is explained by the strong terminator motif, similar to bands near the start of groES/EL and hrcA. To my knowledge, no dnaJ-specific Northern blots have been produced to identify a 1.2kb monocistronic transcript and thus, whether the promoter and observed increase in coverage reflect a true transcription start site remains unknown. Possible alternative explanations include post-transcriptional processing or thermodynamic challenges for the reverse transcriptase\cite{80}. However, the coverage pattern does not show a response to the TEX treatment (\ref{fig:5.17a}) and the former seems unlikely here. In the dnaK/J intergenic region, a decrease in coverage is observed near a strong Rho-independent terminator. Subsequently, increased coverage is observed near a strong promoter motif that might explain previous primer-extension results\cite{80}.


\subsubsection{HrcA Transcript Termination}
The full operon is produced in a 5kb transcript which terminates after the dnaJ gene. In \ref{fig:5.17b}, a dramatic decrease in coverage is observed at the end of the dnaJ gene, 50 bases before the stop codon. Four terminator prediction software did not produce results for the transcription termination region shown here. This location should contain a non-intrinsic termination signal to explain the dramatic decrease in coverage. The residual signal from the nearby ribosomal methyltransferase(CA_C1284) matches well with evidence of a longer 8kb transcript in \textit{B. subtilis}\cite{81}. Given the dramatic decrease in coverage observed here, a reasonable transcription stop site for the 5kb operon can be assumed to follow the dnaJ gene(\ref{fig:5.16c}). Having discussed the regulatory regions of the hrcA operon, the final task is to summarize the transcript lengths, their regulation, and identify missing regulatory elements.


\begin{figure}
\small
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/HrcA/DnaKJ-IGR.png}
\subcaption{DnaK/DnaJ Intergenic Region}\label{fig:5.17a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/HrcA/HrcA-termination.png}
\subcaption{HrcA Transcription Termination Region.}\label{fig:5.17b}}
\caption{HrcA Locus Transcription Termination Regions}
\subref{fig:5.17a}) The dnaK/dnaJ intergenic region consists of a Rho-independent terminator for the 2.6kb and 3.8kb transcripts. The coverage this region rebounds after a promoter motif and is not depleted with TEX treatment. \subref{fig:5.17b}) Near the end of the dnaJ gene, a dramatic decrease in coverage signals the end of the 5kb hrcA operon. No terminators can be found in this region, suggesting a non-intrinsic termination signal.
\end{figure}


\subsubsection{HrcA/GrpE Transcript Lengths}
In the hrcA locus, two transcription initiation regions and two termination regions are responsible for 3 previously reported transcript sizes, 2.6, 3.8, and 5kb, respectively. The longest transcript observed here is 4,838bp from the hrcA transcription initiation region to the termination region downstream from dnaJ. The second largest observable transcript begins at one of the two start sites from the grpE transcription initiation region and ends at dnaJ termination region, leading to transcripts between 3,778 to 4,125 bases, respectively. This region is the most abundant portion of this operon and may contain at least two smaller transcripts. The first of these species would begin at the grpE transcription initiation region and terminate in the dnaK/J intergenic region. This type of transcript would range in size between 2,642bp and 2,989bp, respectively. The fourth and final transcript could originate from the dnaK/J intergenic region and terminate at the end of dnaJ, with a length of approximately 1.2kb. In the model organism \textit{B. subtilis}, larger transcripts are produced from this region, up to 8kb in length\cite{81}. It was observed that the transcripts and proteins of this region increase and decrease at different rates\cite{81}. Determining all the potential transcription start sites inside a region of continuous transcription is an ongoing challenge for understanding of the stress response. To conclude, the regulatory motifs and their locations in this region are summarized.

\subsubsection{HrcA Locus Regulation}
Both Sigma factor A or H dependent promoters could be observed in the transcription initiation regions, as detailed above (\ref{table:2}). Sigma factor A promoters are under the control of the 6S small RNA, which has considerable expression in this study. Therefore, the activity of this locus must be considered with respect to global Sigma A promoter activity. Additionally, the hrcA transcription initiation region is under the direct control of both a CIRCE motif (3.2\e{-13}) and a CtsR motif (1.8\e{-7})\cite{42}. The inverted repeat in the dnaK/J intergenic region does not match either motif and is therefore most likely a Rho-independent terminator. The dnaJ termination region (+/- 200bp from the dnaJ stop codon) does not contain a detectable Rho-independent terminator and thus terminates transcription in a non-intrinsic manner, as previously suggested\cite{83}. The sequencing technique has produced excellent results for this region. The assembly produced a better estimate of the transcription start site than would be expected from coverage alone and a novel transcription start site was identified. Additionally, exact coordinates of regulatory motifs match well with the observed transcription start sites. Finally, potential transcript sizes were discussed to the extent permissible with this technique and the complexity of this region. Next, the spo0A locus is considered an important gene in this organism that has not had the level of molecular investigation of the previous examples.




%          S p o 0 A
\subsection{Spo0A Locus}

\subsubsection{Spo0A}
Spo0A is the master regulator of sporulation and stationary phase phenomena. This protein transduces growth-limiting and stressful signals into sporulation behavior in a number of firmicutes. In previous studies, Spo0A was shown to be translated from a 0.9kb transcript in \textit{C. acetobutylicum}\cite{84}. Additionally, a Sigma factor A and Sigma factor H motif were identified upstream of spo0A, but sadly neither the motifs nor the coordinates were provided. A single Sigma factor A motif (\ref{table:3}) was identified near a substantial increase in transcriptional activity (\ref{fig:5.18a}). A single Spo0A box was reported upstream of spo0A\cite{84} and can be seen in \ref{fig:5.18a}. A single terminator motif is located ~130 basepairs downstream of the spo0A stop codon. The uncurated assembly did not identify single start or stop sites for this gene, fusing spo0A with signal on either side of the gene. After curation with these information combined, we observe a transcript of 1,147bp (\ref{fig:5.18b}), longer than the 0.9kb transcript detected by Northern blot in a previous study of this locus\cite{84}. This represents the first documentation of the Spo0A transcript boundaries in \textit{C. acetobutylicum}.


\begin{table}

%\begin{minipage}[b]{2.5in}
\begin{center}
\begin{tabular}{|c|c|c|c|c|}\hline
\multicolumn{5}{c}{Sigma A}\\\hline
Motif & Start & End & Sequence & p-value\\\hline
-35 & 2173565 & 2173560 & TTGATT & 6.2\e{-3}\\
-10 & 2173537 & 2173542 & TAAAAT & 1.5\e{-3}\\
\hline
\end{tabular}
\end{center}
%\end{minipage}
%\begin{minipage}[b]{2.5in}
\begin{center}
\begin{tabular}{|c|c|c|c|c|}\hline
\multicolumn{5}{c}{Spo0A Box}\\\hline
Motif & Start & End & Sequence & p-value\\\hline
1 & 2173430 & 2173436 & TGTCGAA & 1.9\e{-4}\\
\hline
\end{tabular}
\caption{Spo0A Regulatory Motifs}\label{table:3}
\end{center}
%\end{minipage}
\end{table}

\begin{figure}
\small
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Spo0A/Spo0A-locus.png}
\subcaption{spo0A Locus}\label{fig:5.18a}}
{\includegraphics[width=\textwidth,height=1.5in]{images/Assembly/Examples/Spo0A/Spo0A-curated.png}
\subcaption{Curated spo0A Transcript}\label{fig:5.18b}}
\caption{Spo0A Locus}
\subref{fig:10a}) The spo0A transcript is fused to signal from neighboring genes, although distinct promoter, terminator, and coverage signals are observed. \subref{fig:10b}) After curation, this 1.1kb spo0A transcript reflects the appropriate genomic signals.
\end{figure}


\begin{figure}
\small
\includegraphics[width=\textwidth]{images/Assembly/Examples/Spo0A/CAC2079.png}
\caption{CA_C2079 Gene}\label{fig:5.19}
Slightly downstream of the spo0A gene, a long operon follows a region of high expression on the Crick/minus strand. This coverage pattern is distinct from neighboring regions and surrounds a putative protein that is missing from the databases. This is the first experimental evidence for its expression and it shares homology to efflux transporters.
\end{figure}

\subsubsection{Missing from the Databases: CA_C2079}

In a nearby location, the genes CAC2073-2078 are found in a tight grouping near a large peak of expression (\ref{fig:5.19}). This peak correpsonds to an uncharacterized protein CAC2079 that is present in UniProt but absent from NCBI and KEGG databases. Its substantial sequencing depth appears to be largely above 20k per base, independently transcribed from the surrounding regions. Bioinformatic analysis suggests that it shares sequence similarity to proteins in the \textit{Clostridia}, \textit{Bacilli}, \textit{Baceteroidetes}, and \textit{Halobacteria}. While there was no common catalytic or active domain unifying this group of homologs, a region of homology between them precedes a transmembrane motif. Further analysis via PSI-blast result suggests sequence similarity to mATE (Multidrug And Toxic-compound Extrusion) efflux family proteins. mATE family proteins use electrochemical gradients to export antibiotics and other toxic compounds. The data suggest that the expression of this protein is important to \textit{C. acetobutylicum} and it will be interesting to examine the expression profile statistically. This final example illustrates the synergy of this transcriptomic dataset with existing annotation and the benefit of curation for locating high priority novel transcripts.


In this section, several examples were presented to qualify the sequencing results. After the misassemblies were corrected, the transcript boundaries and sizes were in agreement with previous findings, suggesting precision and accuracy in this technique. In several cases, transcript boundaries required no curation at all. The \textit{sol} locus illustrated the ability of the technique to even multiple transcription start sites with good accuracy. By integrating genomic and transcriptomic signals, the precision of the assembly was improved. After demonstrating the curation technique and validating results with prior studies, the curation technique and results are described in the next section.