% This is the file for the background chapter of my thesis

\chapter{Background}

\section{Climate Change and Renewables: A 21\textsuperscript{st} Century Challege}
Climate change research(reference) and petrochemical exploration(reference) suggest that escalated weather variation, sea levels, and atmospheric and oceanic temperatures will accompany steep increases in the cost of petroleum. Renewable chemical platforms will be an increasingly economical solution to climate change. Renewable biofuels can be carbon neutral sources of energy. Renewable biochemical processes can actually behave as carbon sinks, with net accumulation of CO_2 in chemicals and biomass after subtracting processing energy requirements.

Such a renewable biochemicals systems revolves around a biocatalyst, a microbe that can convert low cost substrates into fuels or chemicals. This production system, frequently referred to as a ``biorefinery,'' requires a microorganism with a wide range of potential feedstocks and a natural biofuel producing metabolism. Biofuels with energy density and hygroscopicity comparable to gasoline and diesel are desirable for infrastructure compatibility. The butanol-producing bacteria \textit{Clostridium acetobutylicum} consumes a number of sugars and hydrolysates and produces a direct gasoline replacement (citation).

\textit{C. acetobutylicum} is a historically industrial solvent producer(reference Weizmann review). It consumes lignocellulose (source), a variety of simple and complex carbohydrates, and hydrolysates. Also known as the Weizmann organism, \textit{C. acetobutylicum} converts these substrates into solvents through an acetone, butanol, and ethanol (ABE) fermentation.  Importantly, these cells synthesize most amino acids with ammonium salts as a nitrogen source, requiring only a minimal defined medium. This microbe meets the requirements for low-cost non-food feedstocks and infrastructure compatibility. Therefore, \textit{C. acetobutylicum} is an excellent chassis organism for an integrated biorefinery and is the system of study in this work.

\textit{C. acetobutylicum} has a number of intrinsic advantages that minimize the engineering efforts required for bioprocess development. It is one of over 17,000 bacteria with sequenced genomes\cite{89,90,91} and has a substantial research community(cite clostridial conference paper?). For example, current metabolic models (cite Maciek's article) are used for sophisticated metabolic analyses, such as $C_{13}$ metabolic flux analysis. A model for the solventogenic \textit{Clostridia}, \textit{C. acetobutylicum} is a reasonably well studied organism with industrial potential.

Prior to the genomic era, targeted studies in this organism revealed the specific loci for solvent formation(references), sporulation(references), and some heat-shock genes(references). These genes were typically cloned, sequenced, and  investigated with gene-specific transcriptomic techniques. However, only the mechanisms of the unique metabolic systems (e.g. solvent formation) have been investigated in detail; The mechanisms for the majority of the genetic and metabolic systems in \textit{C. acetobutylicum} are largely inferred from homology and often, this assumption is appropriate. 

That being said, many of the most interesting characteristics of \textit{C. acetobutylicum} are unique to the \textit{Clostridia}. Of particular interest for biosystems engineering is its solvent stress-response, which may be uniquely adapted to biofuels. A number of stress-response systems exist for specific stresses, with broader systems that respond to multiple stressors. The knowledge of these systems reflects the knowledge of the \textit{C. acetobutylicum} genome generally. In the next section, stress-response systems are reviewed, demonstrating opportunities for the discovery of novel transcriptomic features.

\section{Biofuel/Solvent Tolerance and the Bacterial Stress Response}


Bacteria respond to a wide variety of intrinsic and extrinsic challenges with stress response systems. For example, nutrient deprivation, osmotic shock, temperature fluctuation, and high chemical concentrations are common in their natural environments. Cells activate specific or general response networks after these insults. \textit{C. acetobutylicum} is a naturally solventogenic bacterial species and could possess solvent-tolerance genes that would likely be regulated by general or specific stress response systems. Such genes would be natural targets for biosystems engineering, improving the tolerance of engineered strains to high biofuel titers. 

Unfortunately, knowledge of these systems is incomplete in \textit{C. acetobutylicum} and no unique solvent tolerance genes, such as solvent exporters, have been identified. The stress response is an important system for biotechnology, but the annotations of \textit{Clostridia} genomes require improvement for understanding of these and other systems. Here we review these systems and the extent of their knowledge in \textit{C. acetobutylicum}.


\subsubsection{Specific Stress Response Systems}
Specific stress-responses are designed to mitigate the negative effects of a particular stressor. These systems typically contain a detection mechanism for either the stressor or its effects, such as antibiotics or DNA damage, respectively.

Antibiotic resistance is an example of a specific stress response system that can detect a molecular stressor. In the presence of organic compounds of the $\beta$-lactam, tetracycline, and (others?) families, bacteria activate antibiotic resistance genes that either export or modify the organic compounds to prevent their action (review reference). Some of these antibiotic-resistance genes are part of operons that possess antibiotic-detecting repressors (review citations). Upon the detection of the antibiotic agent, a confirmation change triggers derepression of the antibiotic resistance gene and subsequently an antibiotic resistant phenotype. This stress response system helps the bacteria respond quickly and specifically to compounds from a family of antibiotics, but typically do not confer survival benefits to other stressors, such as heat stress.

A second example of a specific stress response is from \textit{D. radiodurans}, which has an unparalleled resistance to ionizing radiation (reference). In response to breaks in DNA, \textit{D. radiodurans} repairs the multiple copies of its genome, enabling growth, viability, and survival at over 5,000 Gy of radition (reference). This system responds instead to the symptom of $\gamma$-radiation, DNA damage. The extreme tolerance of \textit{D. radiodurans} to radiation is a byproduct of a specialized system for DNA repair, augmented from the standard DNA repair system. This system contains a detection system, non-homologous recombination, and a specialized DNA repair system allows \textit{D. radiodurans} cells to survive and repair hundreds of insults to its genome. 

If specific stress response systems exist for solvent stress (e.g. solvent exporters), they would be desirable targets for research in solventogenic \textit{C. acetobutylicum}. Stress responsive small RNAs have been described(Keerthi's small RNA paper), although their roles, regulation, and conservation remain unknown. To identify genes that could serve this role, differential expression experiments are desirable. The statistical treatment of measurements for such experiments is complex and requires reliable estimates of transcript expression. These estimates would be difficult to acquire with the original bioinformatically-predicted genome annotation. An improved genome annotation could yield novel stress-responsive transcripts and improve expression estimates by counting reads at the transcript level, as opposed to the ORF level.

In addition, an improved genome annotation complete with transcription start sites and regulatory regions would facilitate research on the regulation of a hypothetical specific solvent stress-response system. Regulatory motifs can be discovered with enrichment methods (RSAT reference) with a complete set of transcript boundaries. Regulatory networks are targets for engineering purposes but difficult to research, especially without precise knowledge of gene boundaries.

The first type of stress-response system reviewed can have unique detection and response mechanisms to specific intrinsic or extrinsic stressors. These genes have important applications in environmental remediation (D. raiodurans cleanup), pharmaceutical research (antibiotic research, tier design), and more. The next stress-response systems reviewed respond to more than one stressor, conferring a more general survival benefit.


\subsubsection{General Stress Response Systems}

In contrast to specific stress-response systems, general responses are activated by more than one stimulus. These programs provide non-specific survival benefits to more than one stressor. For example, during both nutrient deprivation and acid stress, cells must slow or cease growth (stringent response) to adapt to energetic demands of the activation of both the specific and general stress response program. Also, both heat-shock(citation) and solvent stress(citation) can denature proteins and consequently activate chaperonin systems, another example of a general stress response. After detection of the stressor, signal transduction events activate dormant response machinery (citation) or activate/derepress response systems(citation). The general stress response is divided into four classes of genes based on the regulator responsible for their activation. 

\paragraph{Class I}
The first class of general stress response genes is governed by the repressor HrcA, which responds to protein denaturation from thermal or chemical causes. The HrcA regulon contains at least 3 transcripts including the HrcA and DnaK/J, GroES/EL, and htpG loci(Qinghua, other citations). Denatured proteins titrate the groEL chaperone from HrcA/groEL complexes, resulting in a conformational change of HrcA and decreased DNA binding(source). Operons regulated by the HrcA repressor are subsequently derepressed, rapidly increasing the amount of heat shock proteins. Protein denaturation negatively affects nearly every program and structure of the cell, resulting in decreased survival and viability. In \textit{C. acetobutylicum}, the HrcA motif was recently described(citation) for standard heat-shock operons. Additionally, class I genes are solvent-stress responsive (citation) due to solvent-induced protein denaturation. It is unknown if any additional operons are also regulated by this repressor in \textit{C. acetobutylicum}. To answer this question, an improved genome annotation including transcription start sites would facilitate the discovery of additional genes in the HrcA regulon through \textit{in silico} analyses.

\paragraph{Class II}
The second class of genes is regulated by a stress-responsive $\sigma$-factor, $\sigma$_{B}. In \textit{B. subtilis}, the $\sigma$_B regulon consists of ___ genes and coordinates a general stress response for a variety of stressors. A \textit{C. acetobutylicum} $\sigma$_B ortholog was not predicted in the initial genome annotation, although the various stress response genes (HrcA under Sig B regulon CITATION) remain. The regulation of these genes is unknown in this organism. Perhaps $\sigma$_{B} genes in \textit{C. acetobutylicum} are under the control of an overlapping regulon, as in \textit{L. monocytogenes}. Given the large size of the $\sigma$_B regulon and the presence of several of its genes (e.g. HrcA(citation), clpC), the promoter and regulatory regions of \textit{C. acetobutylicum} orthologs of $\sigma$_{B} regulon genes would be useful for understanding the unique stress response of \textit{C. acetobutylicum}. This class of general stress-response genes present another opportunity for an improved genome annotation to facilitate stress response research.

\paragraph{Class III}
The third class of genes is regulated by the repressor CtsR, with a ____ transcript regulon in \textit{B. subtilis}. CtsR is a dimeric helix-turn-helix regulator that is capable of responding to heat-shock, oxidative stress, and acid stress(references from lit review of CtsR-thiol paper). In \textit{B. subtilis}, CtsR responds to heat-shock when clpC, the fourth gene of the CtsR operon, releases McsB(reference). Free McsB phosphorylates CtsR, leading to positive autoregulation and derpression of the CtsR regulon. This mechanism is thought to vary across the gram positive bacteria(reference), but the operon organization suggests that the \textit{C. acetobutylicum} CtsR system is similar to \textit{B. subtilis}(Qinghua paper). A CtsR motif was found ahead of canonical CtsR regulon genes in \textit{C. acetobutylicum}(Qinghua paper) although additional genes may be controlled by CtsR in the absence of $\sigma$_{B}. A genome-wide motif search in this organism could similarly reveal CtsR regulation of solvent responsive transcripts. Similar to the class I system, knowledge of the CtsR regulon would also benefit from an improved genome annotation.

\paragraph{Class IV}
The fourth and final class of general stress-response genes are regulated by unknown mechanisms. In \textit{C. acetobutylicuM}, this class includes stress-responsive genes with unconfirmed operon structure and no confirmed motifs. Microarray experiments from Venkataramanan show over 1,000 solvent responsive genes in \textit{C. acetobutylicum}, the regulation of which remains almost completely unknown. As suggested above, an improved genome annotation would aid the categorization of this massive gene set into class I, III, or potentially new regulons.

These general stress response programs are vital for adaptation and robustness. However, both sporulation and stress-response systems in the \textit{Clostridia} differ from the model for sporulating gram positive bacteria, \textit{B. subtilis} (Pap sporulation review). At the very least, specific regulatory genes are missing such as the sporulation regulator ____ (Pap sporulation review) and the stress response regulator $\sigma$_B. Furthermore, most solvent responsive genes in \textit{C. acetobutylicum} are regulated by unknown motifs and response regulators. Finally, it is reasonable to expect that there may be some stress-response genes specific to the solventogenic \textit{Clostridia}, perhaps even exclusively solvent responsive. Such genes would have had little homology to known genes during the initial genome annotation and therefore would not be included in previous comparative genomic and microarray analyses. For example, a recent study identified unique solvent responsive small RNAs in \textit{C. acetobutylicum}, many with no known homologs or regulators (keerthi small RNA). Clearly, there is much to be done to understand and develop this organism for biofuel applications.

In this section, the example of stress-response systems was used to demonstrate differences of \textit{Clostridia} from the \textit{Bacillus} model. Also, it is clear that with current knowledge of the \textit{C. acetobutylicum} genome, there are opportunities to discover novel transcripts or proteins not described by the initial genome annotation and provide precise transcript boundaries for motif identification. In fact, the current annotation relies almost entirely on bioinformatic ORF predictions. To provide definition to the stress response and other systems, regulatory regions and operon organization should be revealed by modern transcriptomic techniques. Next, techniques that provide these details throughout the genome are discussed.

\section{Transcriptomic Research}
Modern high-throughput techniques can produce genome-wide data, even when reference genomes are not available (de novo assembly reference, metatranscriptomics). An enormous number of array and sequencing techniques have been developed to investigate the proteome, transcriptome, and interactome including protein-protein(Chang Tse-Wen), protein-DNA(chip on chip, ChIP-seq), protein-RNA(pulldown, and RNA-RNA interactions (microRNA array, HITS-clip), protein post-translational modifications, single nucleotide polymorphisms, and transcript expression(Mark Schena microarray, RNAseq). These powerful genome-wide techniques allow experimental investigation of nearly every aspect of biological systems, including the identification of all genes and their boundaries. Here, we focus on transcriptomic techniques to identify these features. 

Transcriptomic techniques allow the characterization of the properties and dynamics of RNA species. In the last decade, microarrays and sequencing techniques have revolutionized transcriptomic research, providing new insights into the complexity of the transcriptome (E. coli sRNA, miRNome, ncRNAs, snoRNAs etc) and its regulation(HFQ, Argonaute, turnover). Understanding of this dynamic RNA population is in its infancy, in addition to its interactions with the metabolome(riboswitch) and proteome(HFQ, regulatory proteins) levels. Two common experimental objectives are used to explore cellular programs: annotation and differential expression.

Before differential expression experiments are conducted, or in instances where a complete genome is not available, the catalog of all genes and their transcripts is required for microarray probe design or sequence read counting. ORF prediction algorithms(Glimmer citation, AUGUSTUS) and annotation suites(RAST citation, MG-RAST citation) can identify putative ORFs encoding enzymes and canonical genes for sequenced genomes. However, unique genes and small RNAs cannot be identified with these predictions. It is desirable to experimentally determine the transcriptome for conditions of interest.

Tiling microarrays(Bsub) and deep RNA sequencing(one of the strand specific papers) can be used to identify novel transcripts, transcript boundaries, and operon organization. Strand specific options for these methods are especially useful in dense bacterial genomes. The large amount of data require computational processing to identify the desired features. However, type I and II errors of these techniques make automation of transcript annotation difficult, an issue not addressed by the literature. Nevertheless, transcriptome assembly methods permit the reconstruction of full-length transcripts from sufficiently deep sequencing datasets at the expense of misassembly errors. In this section, transcriptomic techniques are reviewed and experimental designs discussed to identify opportunities for improvement over convention for this study's objective, revealing transcript boundaries, regulatory regions, and operon organization.

\subsection{Analytical Techniques}
\subsubsection{Microarrray}
Microarray-based transcriptomics allows the simultaneous measurement of thousands of sequences simultaneously. Probes designed for genomic sequences or open reading frames(ORFs) measure the expression of segments of the genome or ORFs, respectively. The former, also called tiling microarrays, are used to identify features of the transcriptome and annotate the genome. The latter is a typical practice for expression profiling experiments(cite studies) and identifying relative expression differences. The experienced research community have made the microarray an excellent platform for transcriptomic analyses.

Microarray technology has limitations that should be considered during the early stages of experimental design. First, the microarray suffers from limited detection range and sensitivity(source) compared to RNA sequencing. Additionally, probe design is limited to either ORFs only or to overlapping segments of a genome sequence where available. Finally, the cost of tiling array experiments is a function of the amount of overlap between the probes; higher resolution implies higher cost. The limited detection range could make low-abundance transcripts challenging to identify or distinguish from spurious transcription(spurious transcription citation).

A comprehensive investigation of \textit{B. subtilis} identified many ORFs to be expressed under various environmental and life-cycle conditions(source). The authors detected 85\% of all previously known transcription start sites(cite). With 22-basepair resolution, tiling microarrays were used to detect increases in fluorescent signal across the genome, providing transcript boundaries after manual annotation. The use of manual methods to identify novel genes and transcript boundaries highlights the complexity of these datasets and the lack of methods for automated annotation. This study is an excellent model for transcriptome mapping using the microarray technology.

\subsubsection{RNA Sequencing}
The emerging technology of RNA sequencing(RNA-seq) is displacing tiling arrays for transcriptome mapping and annotation studies. Parallel sequencing platforms such as Illumina(citation) or 454 technology(citation) enable the sequencing of gigabases of cDNA with precision for mapping and counting. cDNA libraries are sequenced in an extremely parallel manner, producing millions of sequences, reads, which are then aligned to the genome. RNA-sequencing has a higher dynamic range than microarray technology in addition to basepair-level resolution. These characteristics make RNA-seq optimal for precise determination of transcript boundaries, even for low abundance transcripts.

To achieve this, each step of the experiment and RNA processing must be considered carefully. Poor consideration of the factors involved with this method leads to uneven coverage (Eco study, transcriptome assembly review, strand specific sequencing rev,other stats reviews), poor depth (Eco study, smallest study), and questionable strand specificity(directional studies). While attempts have been made to establish standards for biomedical RNA-seq (encode reference), no guidelines exist for the broader research community. RNA-seq is frequently performed with the Illumina platform for the amount of data it produces. This amount of data plays a crucial role in detecting transcript boundaries and low abundance transcripts.

A central challenge to RNA-sequencing experiments is related to dynamic range. The likelihood of sampling low-abundance cDNA fragments from transcript termini or low abundance transcripts is a function of the number of sequenced reads(encode, transcriptome assembly, depth vs rep article bioinf). The absence of reads at a location in the genome indicates either an insufficient sequencing depth(type II error) or true absence of transcription. Bacterial cDNA libraries are commonly sequenced in multiplex over one or more lanes of an Illumina sequencer, leading to millions of reads distributed across the libraries. Therefore, there is a three-part tradeoff between the replication, depth, and independent variables in the experimental design. A recent study concerning the tradeoff between the first two concluded that for differential expression experiments, replication is preferable(bioinf rep vs depth). In the case of transcriptome mapping and genome annotation however, additional depth may be preferable(encode).

Standard procedures are required to compare sequencing depth between RNA sequencing studies with similar objectives (e.g. transcription start sites). The Encyclopedia of DNA Elements(ENCODE) research consortium frequently uses RNA-seq(Nature papers) and has published best practice guidelines for Human genome research(citation). While central tendency heuristics and depth distributions are not provided, they suggest that 100-200 million clusters of paired-end reads is sufficient depth to identify novel transcripts and transcriptomic features (e.g. TSSes) in the hg19 \textit{H. sapiens} transcriptome. This number is much larger than required for microbial genomes; a preferable metric is the ratio of the number of reliably-mapped non-ribosomal reads to the approximate size of the transcriptome. Next, this metric is used for persepective to compare transcriptome mapping studies.

\subsection{Transcriptome Mapping Studies and Common Challenges}

\begin{table}
\caption{Study Comparison}\label{table:study_compare}
\begin{center}
\begin{tabular}{|l|c|c|c|c|c|c|c|}\hline
 & Genome(Mbp) & Transcriptome(Mbp) & Clusters(M) & rRNA-free & Reliable & Alignment Rate & Ratio\\\hline\hline
Standard & 3000 & 140 & 30 & 30 & 25 & N/A & 0.18\\\hline
Deep & 3000 & 140 & 100 & 100 & 100 & N/A & 0.71\\\hline
Ultra-deep & 3000 & 140 & 200 & 200 & 200 & N/A & 1.43\\\hline
\textit{C. beijerinckii} & 6 & 6-12 & 14 & N/A & 11.5 & 0.82 & N/A\\\hline
\textit{P. difficile} & 4.3 & 4-8 & 50 & 4 & N/A & N/A & 0.5-1\\\hline
\textit{B. anthracis} & 5.5 & 5-11 & 33 & N/A & 5 & 0.15 & N/A\\\hline
\textit{H. pylori} & 1.7 & 1.5-3 & 0.4 & N/A & 0.2 & 0.54 & N/A\\\hline
\textit{Synechocystis. sp} & 3.9 & 3.9-7.8 & 0.2 & 0.1 & 0.1 & 0.475 & 0.03\\\hline
\textit{E. coli} & 4.6 & 4.5-9 & 52 & N/A & 17.7 & 0.34 & N/A\\\hline
\textit{P. gingivalis} & 2.3 & 2.3-4.5 & 15 & N/A & 2.3 & 0.15 & N/A\\\hline
\textit{S. typhi} & 5 & 5-10 & 5.7 & N/A & 1.8 & 0.31 & N/A\\\hline
\end{tabular}
\end{center}
\end{table}


The most phylogenetically similar organism that has been investigated is \textit{C. beijerinckii}(Citation). In this study, 82\% of the reads aligned uniquely to the genome, a superior alignment rate and depth, considering the previously described ratio, when compared to other studies (table). However, the authors fail to qualify their work with respect to several factors that influence transcript discovery and transcription start site identification. Next, common issues to transcriptome mapping studies are described. 

In dense bacterial genomes, proteins are coded by polycistronic transcripts, operons, that are packed closely in to small circular genomes, only a few megabases in length. In bacteria, overlapping transcripts can be encoded on opposite strands. Strand specific techniques are seldom used(refs), preventing the detection of divergent operons, novel genes in antisense, and \textit{cis}-encoded sRNAs.

Unlike eukaryotic transcriptome mapping studies, bacterial total RNA contains 95-99\% ribosomal RNA. For most RNA-seq applications, the overwhelming presence of rRNA lowers the useful sequencing depth extraordinarily. Several commercial kits are available with suboptimal and inconsistent removal rates(references). Very few studies provided calculations or discussion of the effect that rRNA had on the effective sequencing depth.

Another intrinsic artifact to the RNA sequencing method is the effect of preferential PCR amplification(sources) on library complexity. After cDNA library construction, the library is typically amplified to provide additional material for sequencing. Duplicated reads provide redundant information and can be removed \textit{in silico}(picard reference). According to the methods of each article, no efforts were made to address this issue.  

Ribosomal RNA and PCR-amplification bias are two sources of noise that lower the amount of useful signal from bacterial RNA-seq experiments. \textit{In silico} solutions(bowtie, picard) allow researchers to separate and quantify the useful signal for the purposes of qualification and comparison with other studies. With only partial treatment of these issues in a few studies(ref Synecho. sp and P. diff), the true quantity of data in each was unclear. Specifically, it is unclear what the distribution of per-base sequencing depth actually was; depth of coverage is highly variable in transcriptomic studies. Description of this variable would have helped comparison and experimental design for this work.

The final issue concerns the presence of background signal. Transcriptional noise is a phenomenon that is detectable with high sensitivity methods such as deep RNA sequencing(references). RNA sequencing can detect low abundance signals such as residual genomic DNA, low abundance transcripts, and spurious transcription(refs). It is unclear how this noise was distinguished from signal in these studies, despite low and unstable depth of coverage reported by many studies (Eco, C. beij, others). The effect of these artifacts on type I error is ignored in each of these studies.

These issues highlight the need for standards in RNA sequencing and transcriptome mapping, similar to MIAME standards for microarrays(cite). Of the studies reviewed here, the most depth was achieved in a strand specific experiment for \textit{E. coli}(sources). Even so, the authors report uneven depth of sequencing coverage inside of open reading frames(source) despite the impressive ratio of sequenced reads to the size of the transcriptome(\ref{table:study_compare}). Clearly there are a number of challenges for transcriptome mapping studies, especially with unknown type I and type II error rates typical of exploratory projects. However, by addressing technical obstacles explicitly with bioinformatic methods, the amount of useful data can be quantified. Taken together, these issues were discussed and addressed in this work for similar projects to learn from and improve upon in the future.

\subsection{Lessons and Objectives}
The biofuel producing bacterium \textit{C. acetobutylicum} is an excellent platform organism for bioprocesses. A crucial challenge for productivity in this and other organisms is the tolerance of the host to large biofuel concentrations. Knowledge of the stress response systems of this organism are limited by an antiquated genome annotation(original Cac genome) without transcription start sites, operon organization, or promoter signals. These transcriptomic features are essential for understanding coexpression and regulatory networks. RNA sequencing is a powerful method used here to identify these features for future research on stress response systems and biofuel tolerance. This project aims to identify transcription start sites in the \textit{C. acetobutylicum} genome while addressing frequently ignored issues in sequencing approaches. 

