% This is the Chapter 5 - Conclusions file (conclusions.tex)

\chapter{Conclusions}

Renewables research revolves around the development of a feedstock-flexible chassis organism that requires minimal engineering for biofuels production. \textit{C. acetobutylicum} is a historically industrial organism with these properties. A hurdle to improving productivity in this and other microbes is the understanding of stress response systems and their role in conferring tolerance to high concentrations of fuels. Knowledge of these systems can be obtained through perturbation, differential expression, and ChIP-seq studies, all of which rely on genome annotations. The following examples are essential for the understanding of complex multigenic systems:

\begin{enumerate}
\item Operon structure and rearrangement
\item Coexpression patterns
\item Accurate expression estimates
\item Transcription start sites and promoter/regulatory regions
\item Small RNAs and targets
\end{enumerate}

Objectives of this research included the discovery of novel transcripts and transcript boundaries. These principal results would in turn define operon organization and regulatory regions. To this end, ultra-deep RNA sequencing with the Illumina HiSeq platform\cite{88} provided an absolute measurement technique for the whole transcriptome of \textit{C. acetobutylicum}. A median sequencing depth of 156x indicated the high sensitivity for the detection of low abundance fragments from rare transcripts or terminal 5$\prime$ or 3$\prime$ fragments for precise determination of transcript boundaries. Per-base sequencing depth and alignment statistics suggested that the dataset was high quality and sufficient for the goals of this study.

This high quality dataset was used for transcriptome assembly, which in many cases provides the true transcript boundaries. Many of the bases in the genome were assembled, indicating good k-mer complexity of the reads and a high sensitivity. Type I errors, misassembled bases, resulted from this sensitivity and background signals. A customized genome browser was developed to contextualized the sequencing data with genomic annotations. This integrative analysis increased the signal to noise ratio and improved the precision of the assembly. The final assembly demonstrated single-base resolution transcript boundaries, consistent with previous experiments of canonical loci in \textit{C. acetobutylicum}. The curated assembly of the pSol1 megaplasmid showed improved statistics in agreement with transcriptomic features in \textit{E. coli}. The method and browser was developed for the proof-of-principal determination of 111 transcripts.