% This is the Chapter 2 - Methods file (chap2.tex)

\chapter{Methods}


\section{Culture}
Wild type \textit{Clostridium acetobutylicum} ATCC 824 was cultured anaerobically in 4L New Brunswick Scientific BioFlo 310 bioreactors at \SI{37}{\degreeCelsius}, pH >5.0, \SI{200}{\milli\liter\per\minute} N\textsubscript{2} and 200rpm agitation in a defined \textit{Clostridia} growth medium, as described previously (Venkataramanan 2013). When the cultures were grown to A\textsubscript{600}=1, the N\textsubscript{2} flow rate was decreased to \SI{50}{\milli\liter\per\minute} and cultures were either stressed to a final concentration of \SI{60}{\milli\Molar} \textit{n}-butanol, \SI{40}{\milli\Molar} potassium butyrate, or left unstressed. \SI{15}{\milli\liter} samples were acquired at 15, 75, 150, and 270 minutes after treatment and synchronization. Samples were centrifuged at 8,000rpm, \SI{4}{\degreeCelsius} for 20 minutes. After discarding the supernatant, cell pellets were then immediately frozen at \SI{-85}{\degreeCelsius}.

\section{RNA preparation}
RNA was extracted by first washing the cell pellets in 1mL of RNase-free SET buffer (25\% sucrose, \SI{50}{\milli\Molar} EDTA [pH 8.0], \SI{50}{\milli\Molar} Tris-HCl [pH 8.0]) before resuspending cells in a \SI{220}{\milli\liter} solution of RNase-free SET buffer containing \SI{4.55}{\Units\per\milli\liter} proteinase K and \SI{20}{\milli\gram\per\milli\liter} lysozyme and incubating for 6 minutes. Resuspended cells were vortexed with 40mg of RNase-free glass beads ($\leq$\SI{106}{\micro\metre}) at maximum speed and room temperature for 4 minutes. Each sample was mixed immediately with \SI{1}{\milli\liter} of ice-cold QIAzol (Qiagen, Valencia, CA, USA) and then \SI{200}{\micro\liter} of ice-cold chloroform, mixing well. After a 3 minute room temperature incubation, samples were centrifuged at 11,000rpm and \SI{4}{\degreeCelsius} for 15 minutes. The aqueous phase was then mixed with \SI{1.3}{\milli\liter} of ice-cold ethanol before transfering to a miRNeasy Mini spin-column (Qiagen, Valencia, CA, USA) and centrifuging at 11,000rpm and \SI{4}{\degreeCelsius} for 15 seconds.

Next, \SI{700}{\micro\liter} of RWT buffer was added to the column, before centrifuging at 11,000rpm and \SI{4}{\degreeCelsius} for 15 seconds, discarding the collection tube and transferring the column to a fresh collection tube. The column was washed twice with \SI{500}{\micro\liter} of RPE buffer before centrifuging at 11,000rpm and \SI{4}{degreeCelsius} for 15 seconds each. The membrane was then dried with an additional centrifugation step at 11,000rpm and \SI{4}{degreeCelsius} for 1 minute. The RNA was eluted twice by incubating with \SI{50}{\micro\liter} of nuclease-free water for 1 minute and eluting for 1 minute at 11,000rpm and \SI{4}{degreeCelsius}.

After quantification on a Nanodrop ND-1000, samples were then precipitated in 0.3M sodium acetate and 75\% ethanol overnight, centrifuged at 14,000 rpm for 30 minutes, washed twice with \SI{400}{\micro\liter} ice-cold 70\% ethanol, and rehydrated in \SI{50}{\micro\liter} RNase-free water. Next, samples were treated with the Turbo DNA-free kit (Ambion, Austin, TX, USA). \SI{5}{\micro\liter} of 10X Turbo DNase buffer and \SI{1}{\micro\liter} of Turbo DNase (2U\SI{}{\per\micro\liter}) were added to each sample before incubating at \SI{37}{degreeCelsius} for 30 minutes. Next, \SI{5}{\micro\liter} of DNase inactivation reagent were added to each sample, mixing occasionally for 5 minutes. The samples were then centrifuged at 10,000rpm and \SI{4}{degreeCelsius} for 90 seconds, precipitating the DNase. 

Samples were then precipitated, washed twice more with 70\% ethanol, and resuspended in \SI{20}{\micro\liter} of nuclease-free water, requantified, and aliquoted for quality analysis with the BioAnalyzer platform (Agilent, Wilmington, DE, USA), and \SI{10}{\micro\gram} aliquots in \SI{10}{\micro\liter} samples were stored at \SI{-85}{\degreeCelsius}.

\section{RNA enrichment, RNA-seq library preparation, and Sequencing}
Ribosomal RNA was removed with the MicrobExpress kit (Ambion, Austin, TX, USA) according to their protocol. Briefly, beads were prepared by taking \SI{50}{\micro\liter} for each sample, washing with an equal volume (\SI{50}{\micro\liter}) of water capturing for 5 minutes on a MagnaSphere (Promega, Madison, WI, USA) magnetic stand and aspirating. Subsequently, the beads were resuspended in an equal volume (\SI{50}{\micro\liter} each) of binding buffer and capturing as above. The beads were then resuspended in an equal volume (\SI{50}{\micro\liter} each) of binding buffer and warmed to \SI{37}{\degreeCelsius}. Next, \SI{200}{\micro\liter} of binding buffer was added to each \SI{10}{\micro\gram} RNA aliquot with \SI{4}{\micro\liter} of capture oligo mix. The mixture was warmed to \SI{70}{\degreeCelsius} for 10 minutes, then cooled to \SI{37}{\degreeCelsius} for 15 minutes. Next, the rRNA was captured by mixing \SI{50}{\micro\liter} of beads with each sample, incubating for 15 minutes at \SI{37}{\degreeCelsius}, and capturing as above. The enriched RNA was transferred to a fresh \SI{1.5}{\milli\liter} tube. The beads were then washed with \SI{100}{\micro\liter} of pre-warmed (\SI{37}{\degreeCelsius}) wash solution, incubating on the magnetic stand for 5 minutes, and adding the wash solution to the enriched RNA. The samples were then ethanol precipitated at \SI{20}{\degreeCelsius} overnight with \SI{35}{\micro\liter} of \SI{3}{\Molar} Sodium Acetate, \SI{5}{\milli\gram\per\milli\liter} Glycogen, and \SI{1175}{\micro\liter} of chilled 100\% ethanol. The samples were washed twice with 70\% ethanol and resuspended in \SI{25}{\micro\liter}. The samples were enriched further by repeating the MicrobExpress treatment. Small 10-\SI{100}{\nano\gram} aliquots were analyzed at each step with the BioAnalyzer to monitor enrichment.

Selected samples were enriched further with Terminator 5'-phosphate dependent exonuclease kit (Epicentre, Madison, WI, USA). Terminator Exonuclease \SI{1}{\micro\liter} (1U\SI{}{\per\micro\liter}) was added with \SI{2}{\micro\liter} 10X Buffer A to each RNA sample. The reaction was run in a      thermocycler for 60 minutes at \SI{30}{\degreeCelsius}. The reaction was terminated with the addition of \SI{1}{\micro\liter} of \SI{100}{\milli\Molar} EDTA and ____ Tris HCl at pH 8.0. The samples were then purified by ethanol precipitation (\SI{0.3}{\Molar} Sodium Acetate and 75\% ethanol) with two 70\% ethanol washes, as above.
Enriched RNA was quantified as above and assessed for quality with the BioAnalyzer platform (Agilent, Wilmington, DE, USA). High quality samples were used to prepare RNA-seq libraries with the ScriptSeq v2 library preparation kit and indexed PCR primers (Epicentre, Madison, WI, USA). Briefly,  \SI{1}{\micro\liter} of fragmentation solution and \SI{2}{\micro\liter} of cDNA synthesis primer was added to \SI{50}{\nano\gram} of RNA and the solution was fragmented for 5 minutes at \SI{85}{\degreeCelsius} in a _____ thermocycler. To each reaction, \SI{0.5}{\milli\Molar} of Dithiothreitol, \SI{3}{\micro\liter} of cDNA synthesis premix, \SI{0.5}{\micro\liter} StarScript Reverse Transcriptase. is added to each sample and run with the following cycle: 5 minutes at \SI{25}{\degreeCelsius}, 20 minutes at \SI{42}{\degreeCelsius}. After cooling each reaction to \SI{37}{\degreeCelsius}, \SI{1}{\micro\liter} of finishing solution was added, incubating for 10 minutes. The RNA is degraded by fragmenting further for 3 minutes at \SI{95}{\degreeCelsius}, cooling to \SI{25}{\degreeCelsius}. The first strand cDNA is di-tagged by adding \SI{7.5}{\micro\liter} of terminal tagging premix and \SI{0.5}{\micro\liter} of DNA polymerase. The terminal tagging reaction is run at \SI{25}{\degreeCelsius} for 15 minutes and \SI{95}{\degreeCelsius} for 3 minutes. The di-tagget cDNA is then purified with the AMPure XP bead system (Beckmann Coulter, Brea, CA, USA). First, the library is mixed with \SI{45}{\micro\liter} of homogenous bead mixture. After thorough mixing, each solution is transferred to a \SI{1.5}{\milli\liter} tube and the library is captured with the magnetic stand and the supernatant aspirated. Each library is then washed twice with \SI{200}{\micro\liter} of 80\% ethanol. After resuspending in \SI{24.5}{\micro\liter} of nuclease-free water, the beads are captured and each library is transferred to a new \SI{200}{\micro\liter} microfuge tube. Adapters are added to the di-tagged cDNA during PCR by adding \SI{25}{\micro\liter} FailSafe Premix E, \SI{1}{\micro\liter} forward primer, \SI{1}{\micro\liter} of ScriptSeq v2 indexed reverse PCR primer, \SI{0.5}{\micro\liter} of FailSafe Polymerase. The PCR conditions are as follows: cycles of 30 seconds of \SI{95}{\degreeCelsius}, 30 seconds of \SI{55}{\degreeCelsius}, and 3 minutes of \SI{68}{\degreeCelsius}. After 12 cycles, the reaction terminates with a 7 minute incubation at \SI{68}{\degreeCelsius} before purifying the library with the AMPure system, as above.  Libraries were multiplexed and sequenced for 101 cycles over two lanes of an Illumina HiSeq 2500 at the University of Delaware Sequencing and Genotyping Center (Newark, DE, USA).
\section{Sequencing Statistics, Alignments}
Paired-end sequencing resulted in ____ pairs of ___bp reads which are deposited in the Sequence Read Archive ( _____ ). Summary statistics for the libraries are shown in table/appendix ( ___ ). The basic bioinformatic processing pipeline is described on \href{https://github.com/mrals89/NGS_scripts/tree/paired}{Github}. In brief, adaptors are removed in a pair-aware manner from the reads with Trimmomatic. Redundant reads are removed, that is reads from the same cluster that contain identical information. Base quality is adjusted by trimming to the minimum base quality of 20. Before aligning to the \textit{Clostridium acetobutylicum} ATCC 824 genome, the reads are processed for \textit{in silico} ribosomal RNA removal. The reads are aligned to the rRNA sequences with Bowtie 2.1.0. The unmapped reads are then aligned to the genome and megaplasmid sequences (NC_003030.1 and NC_001988.2). The alignment files were then cleaned, sorted, indexed, and validated with SAMtools and Picard.

\section{Transcriptome Assembly}
Reference transcriptome assembly was performed with the Cufflinks suite (v. 2.2.0) and the April 13th, 2014 patch of Trinity. A \textit{de novo} assembly was done with Trinity as well. This assembly was assessed by ____. The number of singletons was assessed by counting the number of reads left unmapped from the assembly.

The coverage vectors for each strand was calculated by assembling proper pairs into fragments in a BED file, and calculating coverage with BEDtools. Transcriptional start sites were identified with with the peak-finding algorithms TSSi. The fully merged transcriptome assembly were used to determine digital gene expression for visualization and statistical analyses in R and Circos. PCA and clustering was performed in R with          and promoter predictions were performed with MEME, and RSAT.
\section{Molecular Methods}
Fold changes were confirmed for randomly selected genes by qRT-PCR. Transcriptional start sites were verified by 5\textprime  RACE for randomly selected genes. Small RNAs were confirmed by RT-PCR and Northern blot.
