% This is the Chapter 2 - Methods file (chap2.tex)

\chapter{Methods}


\section{Culture}
Wild type \textit{Clostridium acetobutylicum} ATCC 824 was cultured anaerobically in 4L New Brunswick Scientific BioFlo 310 bioreactors at \SI{37}{\degreeCelsius}, pH 5.0, \SI{200}{\milli\liter\per\minute} N\textsubscript{2} and 200rpm agitation in a defined \textit{Clostridia} growth medium, as described previously (Venkataramanan 2013). When the cultures were grown to A\textsubscript{600}=1, the N\textsubscript{2} flow rate was decreased to \SI{50}{\milli\liter\per\minute} and cultures were stressed to a final concentration of \SI{90}{\milli\Molar} \textit{n}-butanol, \SI{48}{\milli\Molar} potassium butyrate, or left unstressed. \SI{15}{\milli\liter} samples were acquired at 15, 75, 150, and 270 minutes after treatment and centrifuged at 8,000rpm, \SI{4}{\degreeCelsius} for 20 minutes. After discarding the supernatant, cell pellets were then immediately frozen at \SI{-85}{\degreeCelsius}.

\section{RNA preparation}
RNA was extracted by first washing the cell pellets in 1mL of RNase-free SET buffer (25\% sucrose, \SI{50}{\milli\Molar} EDTA [pH 8.0], \SI{50}{\milli\Molar} Tris-HCl [pH 8.0]) before resuspending cells in a \SI{220}{\milli\liter} solution of RNase-free SET buffer containing \SI{4.55}{\Units\per\milli\liter} proteinase K and \SI{20}{\milli\gram\per\milli\liter} lysozyme and incubating for 6 minutes. Resuspended cells were vortexed with 40mg of RNase-free glass beads ($\leq$\SI{106}{\micro\metre}) at maximum speed and room temperature for 4 minutes. Then each sample was mixed with \SI{1}{\milli\liter} of ice-cold QIAzol (Qiagen, Valencia, CA, USA) and then \SI{200}{\micro\liter} of ice-cold chloroform. After a 3 minute room temperature incubation, samples were centrifuged at 11,000rpm for 15 minutes at \SI{4}{\degreeCelsius}. The aqueous phase was then mixed with \SI{1.3}{\milli\liter} of ice-cold ethanol before transfering to a miRNeasy Mini spin-column (Qiagen, Valencia, CA, USA), following the manufacturers instructions. Samples were then precipitated in 0.3M sodium acetate and 75\% ethanol overnight, centrifuged at 14,000 rpm for 30 minutes, washed with ice-cold 70\% ethanol, and rehydrated in RNase-free water. Next, samples were treated with the Turbo DNA-free kit (Ambion, Austin, TX, USA) according to manufacturers instructions. Samples were then precipitated, washed, and quantified with a Nanodrop ND-1000, aliquoted for quality analysis with the BioAnalyzer platform (Agilent, Wilmington, DE, USA), and \SI{10}{\micro\gram} aliquots in \SI{10}{\micro\liter} samples were stored at \SI{-85}{\degreeCelsius}.
\section{RNA concentration and RNA-seq library preparation}
Ribosomal RNA was removed with the MicrobExpress kit (Ambion, Austin, TX, USA). Selected samples were enriched further with Terminator 5'-phosphate dependent exonuclease kit (Epicentre, Madison, WI, USA). Enriched RNA was assessed for quality with the BioAnalyzer platform (Agilent, Wilmington, DE, USA). High quality samples were used to prepare RNA-seq libraries with the ScriptSeq v2 library preparation kit and indexed PCR primers (Epicentre, Madison, WI, USA). Libraries were multiplexed and sequenced for 101 cycles over two lanes of an Illumina HiSeq 2500 at the University of Delaware Sequencing and Genotyping Center (Newark, DE, USA).
\section{Data Analysis}
Paired-end sequencing resulted in             pairs of 100bp reads which are deposited in the Sequence Read Archive (          ). Summary statistics for the libraries are shown in table/appendix (        ). The basic bioinformatic processing pipeline is described on \href{https://github.com/mrals89/NGS_scripts/tree/paired}{Github}. In brief, after removing adapters and low quality bases with Trimmomatic, reads were first mapped to the ribosomal RNA sequences before mapping to the \textit{C. acetobutylicum} genome both with Bowtie 2.1.0 [parameters:                ]. The resulting alignment was then cleaned, validated, sorted, and indexed with SAMtools and Picard. Unmapped and duplicate reads were flagged for omission from further analysis. Reference transcriptome assembly was performed with the Cufflinks suite (v. 2.2.0) or Trinity. The coverage vectors for each strand was calculated by assembling proper pairs into fragments in a BED file, and calculating coverage with BEDtools. Transcriptional start sites were identified with with the peak-finding algorithms TSSi. The fully merged transcriptome assembly were used to determine digital gene expression for visualization and statistical analyses in R and Circos. PCA and clustering was performed in R with          and promoter predictions were performed with MEME, and RSAT.
\section{Molecular Methods}
Fold changes were confirmed for randomly selected genes by qRT-PCR. Transcriptional start sites were verified by 5\textprime  RACE for randomly selected genes. Small RNAs were confirmed by RT-PCR and Northern blot.
