
%%% Local Variables: 
%%% mode: latex
%%% TeX-master: "main"
%%% End: 
\section{Resolving Misassembly}

These examples illustrated some common themes for misassemblies throughout the genome, notably the effect of background signal from high depth sequencing. Three types of misassemblies were described: extension, fusion, and truncation. In all cases these errors were resolved by solved by combining signals from sequencing depth and genome annotations. The assembly was fully curated for all pSol1 transcripts (the pSol1 megaplasmid is 5\% of the genome). This section explicitly states the rules and heuristics used for assembly curation. Also, the assembly was analyzed before and after curation to determine the effectiveness of this technique. 

The curation process was used in the previous section to address the misassemblies, providing precision transcript boundary estimates. This technique used heuristics to correct transcript boundaries according to depth and annotations in cases where interfering signals result in misassembly. Background signals were defined as extended (typically intergenic or antisense) regions of low depth that frequently conflicted with matching promoter/terminator annotations and large depth fold changes near transcript termini. Briefly, the heuristics are as follows:

\begin{enumerate}
\item Weak terminators ($\delta$G \textgreater -6kcal/mol) were omitted from analysis.
\item Weak promoters and TFBSes were excluded from analysis when p \textgreater 1\e{-5}, defined below where p is defined below for upstream and downstream motif matches (e.g. -35 and -10 elements of $\sigma$_{A} promoter). 
\[ p_{promoter} = p_{upstream} \times p_{downstream} \]
\item Extended transcripts were corrected by shortening or spliting transcripts, such that the resulting transcript(s) captured depth patterns and annotations.
\item Fused transcripts were similarly addressed, with terminators as an important signal.
\item Truncated transcripts typically accompanied obvious trends in depth (e.g. BdhA) and were corrected by extension of the transcript to termini suggested by both depth and genome annotation.
\item Almost always, two or more signals (i.e. depth and terminator, etc.) in agreement were used to determine the true transcript boundaries. In edge cases, extended location specific differences in depth (\textgreater 2 fold change) consistent across replicates were determined to be a transcript terminus.
\end{enumerate}

\begin{table}
\begin{center}
\begin{tabular}{|c|c|c|}\hline
  & Uncurated & Curated\\\hline\hline
Transcripts & 181 & 111\\\hline
Sequenced Mb & 347413 & 192069\\\hline
Length Range & 202-16389 & 172-11397\\\hline
CDSes & 155 & 157\\\hline
Standard Transcripts & 59 & 86\\\hline
Standard Mb & 246926 & 174801\\\hline
Novel Transcripts & 122 & 24\\\hline
Novel Mb & 100487 & 14994\\\hline
\end{tabular}
\end{center}
\caption{Final Assmelby Statistics and Curation Effect}\label{table:assemb_curation}
This table shows final assembly statistics and the corrective power of the curation method. The number of misassembled baspairs and transcripts has been substantially reduced. Two additional ORFs/CDSes were included upon curation and a number of standard transcripts were split in half. Interestingly, about 20\% of the assembled transcripts were novel, a good number of interesting candidates from a small portion of the total \textit{C. acetobutylicum} genome.
\end{table}



These heuristics guided the curation process, addressing the errors described in previous sections, similarly to the treatment of the example transcripts. The most common types of misassemblies were the result of residual background signal assembled and mixed with true transcriptomic signal. The three types of errors were corrected during curation of the entire 192kb pSol1 megaplasmid, resulting in 111 transcripts spanning 192kb (\ref{table:assemb_curation}). In addition to improving the precision of transcript boundary determination, the type I error for transcript discovery was reduced by removing a large number of false positive transcripts.

\begin{figure}
\begin{center}
\begin{minipage}{.48\textwidth}
\begin{center}
{\includegraphics[width=\linewidth,height=3in]{images/Assembly/Curation/PairvsCuration_utrlength.png}
\subcaption{5', 3', and Intraoperonic Untranslated Region Lengths}\label{fig:5.20a}}
\end{center}
\end{minipage}
\begin{minipage}{.48\textwidth}
\begin{center}
{\includegraphics[width=\linewidth,height=3in]{images/Assembly/Curation/PairvsCuration_igrlength.png}
\subcaption{Intragenic Region Lengths}\label{fig:5.20b}}
\end{center}
\end{minipage}%
\end{center}
\caption{UTR and IGR Lengths}
\subref{fig:5.20a}) UTR lengths were considerably improved by curation, with most less than 100bp, in agreement with \textit{E. coli} averages.\cite{87} Interestingly, a number of large 3' and intraoperonic regions remain after curation, suggesting either regulatory roles or the presence of unannotated proteins.

\subref{fig:5.20b}) The size of intragenic regions (IGRs) increased upon curation after the drastic reduction in false-positive basepairs (\ref{table:assemb_curation}).
\end{figure}

The transcript length distributions agreed with prokaryotic averages (\ref{fig:5.22}) after improved precision detailed above.\cite{86} Specifically, the distribution of untranslated region lengths closely matched \textit{E. coli} averages\cite{87} (\ref{fig:5.20a}). In contrast to the decreased transcript and UTR lengths, intergenic region lengths increased upon curation (\ref{fig:5.20b}). The curated transcripts are evenly spaced along the pSol1 megaplasmid, in contrast to the uncurated assembly. Additionally, both the standard and novel transcripts have higher average per-base depth after curation, comparable to the coverage of the reference ORFs (\ref{fig:5.21}). These data show considerable agreement with \textit{E. coli} averages and improvement over the uncurated assembly results.



\begin{figure}
\small
\includegraphics[width=\textwidth]{images/Assembly/Curation/CurvsUncur_boxplot.png}
\caption{Cumulative Depth Distribution}\label{fig:5.21}
After curation, the expression level as indicated by the per-base sequencing depth has increased substantially. While uncurated novel transcripts (far-left) had an order of magnitude lower average sequencing depth than found in reference ORFs (far-right, the 24 curated novel transcripts (center) had comparable sequencing depth. The depth of uncurated standard transcripts (middle left) were only slightly improved in terms of depth by curation (middle right). The best improvement in the novel transcripts was the increased precision of the final transcript boundaries.
\end{figure}

\begin{wrapfigure}{R}{0.4\textwidth}
\small
\vspace{-20pt}
\begin{center}
\includegraphics[width=\linewidth,height=3in]{images/Assembly/Curation/PairvsCuration_length.png}
\caption{Transcript Lengths}\label{fig:5.22}
The transcript lengths were improved after curation, centering the distribution on the standard transcripts on the \textit{E. coli} average of 1.1kb\cite{86}.
\end{center}
\vspace{-20pt}
\end{wrapfigure}
